### PR TITLE
Generate converter files needed for the build with Bazel

### PR DESCRIPTION
**Pull request by mattsoulanille**
Generate `src/operations/op_list/` ops from JSON files in `python/tensorflowjs/op_list`.

Generate `metadata/kernel2op.json` from the source `.ts` files using Bazel.

To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/5393)

### CODE REVIEW

File: tfjs-converter/scripts/gen_op.ts [hunk 1]
```diff
@@ -1,8 +1,24 @@
-import {OpMapper} from '../types';
```
Comment (mattsoulanille) [inline tfjs-converter/scripts/gen_op.ts:1]: This is a new file. I'm not sure why Git thought it was moved.

File: tfjs-converter/src/metadata_test.ts [hunk 2]
```diff
@@ -22,7 +22,7 @@ describe('kernel2op metadata file', () => {
   it('has kernel2op.json', () => {
     expect(() => {
       // tslint:disable-next-line:no-require-imports
-      require('../metadata/kernel2op.json');
+      require('tfjs-converter/metadata/kernel2op.json');
```
Comment (mattsoulanille) [inline tfjs-converter/src/metadata_test.ts:25]: This import breaks with `../` when we use the generated op files (or any generated files for that matter), and I'm not really sure why. Using an absolute import works.

File: tfjs-converter/src/version_test.ts [hunk 3]
```diff
@@ -14,17 +14,17 @@ describe('tfjs-core version consistency', () => {
   it('dev-peer match', () => {
     const tfjsCoreDevDepVersion =
         // tslint:disable-next-line:no-require-imports
-        require('../package.json').devDependencies['@tensorflow/tfjs-core'];
+        require('tfjs-converter/package.json').devDependencies['@tensorflow/tfjs-core'];
```
Comment (mattsoulanille) [inline tfjs-converter/src/version_test.ts:17]: Same here as in `metadata_test.ts`. Using any generated files breaks this import for some reason.

File: tfjs-core/src/version_test.ts [hunk 4]
```diff
@@ -20,7 +20,7 @@ import {version_core} from './index';
 describe('version', () => {
   it('version is contained', () => {
     // tslint:disable-next-line:no-require-imports
-    const expected = require('../package.json').version;
+    const expected = require('tfjs-core/package.json').version;
```
Comment (mattsoulanille) [inline tfjs-core/src/version_test.ts:23]: Same as the other cases that use generated files.

**Comment by mattsoulanille** [comment]
I used git's reviewer for this PR since the Reviewable button hadn't appeared yet, and I assumed it wouldn't since this isn't a PR into master. Feel free to switch to Reviewable.

**Comment by mattsoulanille** [comment]
Some of the python tests are failing, and while that doesn't matter for this PR, that stops GCP from testing downstream packages. Here's a run with them disabled: https://console.cloud.google.com/cloud-build/builds/d06227f4-1dd5-4ee6-859a-b3f78341c63e?project=834911136599

**Review by pyu10055 — CHANGES_REQUESTED** [review]
Thanks, it would be good to update the op contribution doc to indicate json will be source of true instead of the ts file.
https://github.com/tensorflow/tfjs/blob/9370884478680e18ab7c916dfdc0e409bb6d62fc/CONTRIBUTING_MISSING_OP.md

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/5393#-:-Mfi8WvmAXQvJQ1ArtFt:b-1rfy98)* status: 0 of 1 approvals obtained (waiting on @mattsoulanille)

___
*[tfjs-converter/src/BUILD.bazel, line 17 at r2](https://reviewable.io/reviews/tensorflow/tfjs/5393#-Mfi778m4r70cZRUQuBE:-Mfi778m4r70cZRUQuBF:b-2znif4) ([raw file](https://github.com/tensorflow/tfjs/blob/433e8be174b0aed6f6d974047a4d303deab26bc5/tfjs-converter/src/BUILD.bazel#L17)):*
> ```Starlark
> 
> load("//tfjs-converter/scripts:gen_op.bzl", "gen_op")
> load("//tfjs-converter/scripts:ops.bzl", "OPS")
> ```

is it possible to not declare op category list, since it should be derivable from the json file list. We could add more categories in the future, we will need to remember to update this list or there will be build error if there are mismatches?

**Review by mattsoulanille — COMMENTED** [review]
Thanks for the comments. I've updated the contribution doc and made it so we don't have a static list of ops.

When you have tests passing on your branch, I can rebase and see if they still pass with these changes.

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/5393#-:-MfiZz4S3bU1Y_qpyR6q:bxjon64)* status: 0 of 1 approvals obtained (waiting on @mattsoulanille and @pyu10055)

___
*[tfjs-converter/src/BUILD.bazel, line 17 at r2](https://reviewable.io/reviews/tensorflow/tfjs/5393#-Mfi778m4r70cZRUQuBE:-MfiZdBKBU18K2szDBqH:bb4z60b) ([raw file](https://github.com/tensorflow/tfjs/blob/433e8be174b0aed6f6d974047a4d303deab26bc5/tfjs-converter/src/BUILD.bazel#L17)):*
<details><summary><i>Previously, pyu10055 (Ping Yu) wrote…</i></summary><blockquote>

is it possible to not declare op category list, since it should be derivable from the json file list. We could add more categories in the future, we will need to remember to update this list or there will be build error if there are mismatches?
</blockquote></details>

In effect, yes. I made the rule take a glob filegroup of all the .json op files so contributors don't have to edit a .bzl file when adding a new one.

**Review by pyu10055 — APPROVED** [review]
> Reviewed 31 of 40 files at r1, 3 of 7 files at r2, 7 of 7 files at r3.
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/5393)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @mattsoulanille)

**Pull request merged**